src/Compilers/CSharp/Test/Semantic/Semantics/Utf8StringsLiteralsTests.cs

```diff
@@ -0,0 +1,1790 @@
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        var a = ((byte[] b, (byte[] d, string e) c))(""hello"", (""dog"", ""cat""));
+        System.Console.WriteLine();
+        Helpers.Print(a.b);
+        Helpers.Print(a.c.d);
+    }
+}
+";
+            var comp = CreateCompilation(source + HelpersSource, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe);
+
+            CompileAndVerify(comp, expectedOutput: @"
+{ 0x68 0x65 0x6C 0x6C 0x6F }
+{ 0x64 0x6F 0x67 }
+").VerifyDiagnostics();
+
+            comp = CreateCompilation(source + HelpersSource, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe, parseOptions: TestOptions.RegularNext);
+
+            CompileAndVerify(comp, expectedOutput: @"
+{ 0x68 0x65 0x6C 0x6C 0x6F }
+{ 0x64 0x6F 0x67 }
+").VerifyDiagnostics();
+
+            comp = CreateCompilation(source + HelpersSource, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe, parseOptions: TestOptions.Regular10);
+            comp.VerifyDiagnostics(
+                // (7,54): error CS8652: The feature 'Utf8 String Literals' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         var a = ((byte[] b, (byte[] d, string e) c))("hello", ("dog", "cat"));
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, @"""hello""").WithArguments("Utf8 String Literals").WithLocation(7, 54),
+                // (7,64): error CS8652: The feature 'Utf8 String Literals' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         var a = ((byte[] b, (byte[] d, string e) c))("hello", ("dog", "cat"));
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, @"""dog""").WithArguments("Utf8 String Literals").WithLocation(7, 64)
+                );
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void InvalidContent_01()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        byte[] array = ""hello \uD801\uD802"";
+    }
+}
+";
+            var comp = CreateCompilation(source + HelpersSource, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe);
+            comp.VerifyEmitDiagnostics(
+                // (7,24): error CS8983: The input string cannot be converted into the equivalent UTF8 byte representation. Unable to translate Unicode character \\uD801 at index 6 to specified code page.
+                //         byte[] array = "hello \uD801\uD802";
+                Diagnostic(ErrorCode.ERR_CannotBeConvertedToUTF8, @"""hello \uD801\uD802""").WithArguments(@"Unable to translate Unicode character \\uD801 at index 6 to specified code page.").WithLocation(7, 24)
+                );
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void MissingHelpers_01()
```

Comment:
I found the test name slightly confusing because of the `Helpers` class in the test code. Consider renaming to something like `MissingWellKnownMember` or similar.